packages/block-editor/src/hooks/auto-inserting-blocks.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addFilter } from '@wordpress/hooks';
+import { Fragment } from '@wordpress/element';
+import { PanelBody, ToggleControl } from '@wordpress/components';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { createBlock, store as blocksStore } from '@wordpress/blocks';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { InspectorControls } from '../components';
+import { store as blockEditorStore } from '../store';
+
+function AutoInsertingBlocksControl( props ) {
+	const autoInsertedBlocksForCurrentBlock = useSelect(
+		( select ) => {
+			const { getBlockTypes } = select( blocksStore );
+			return getBlockTypes()?.filter(
+				( block ) =>
+					block.autoInsert &&
+					Object.keys( block.autoInsert ).includes( props.blockName )
+			);
+		},
+		[ props.blockName ]
+	);
+
+	const {
+		autoInsertedBlockClientIds,
+		blockIndex,
+		rootClientId,
+		innerBlocksLength,
+	} = useSelect(
+		( select ) => {
+			const { getBlock, getBlockIndex, getBlockRootClientId } =
+				select( blockEditorStore );
+			const _rootClientId = getBlockRootClientId( props.clientId );
+
+			const _autoInsertedBlockClientIds =
+				autoInsertedBlocksForCurrentBlock.reduce(
+					( clientIds, block ) => {
+						const relativePosition =
+							block?.autoInsert?.[ props.blockName ];
+						let autoInsertedBlock;
+						if (
+							[ 'before', 'after' ].includes( relativePosition )
+						) {
+							const siblings =
+								getBlock( _rootClientId )?.innerBlocks;
+
+							// Any of the current block's siblings (with the right block type) qualifies
+							// as an auto-inserted block (inserted `before` or `after` the current one),
+							// as the block might've been auto-inserted and then moved around a bit by the user.
+							autoInsertedBlock = siblings.find(
+								( { name } ) => name === block.name
+							);
+						} else if (
+							[ 'first_child', 'last_child' ].includes(
+								relativePosition
+							)
+						) {
+							const { innerBlocks } = getBlock( props.clientId );
+
+							// Any of the current block's child blocks (with the right block type) qualifies
+							// as an auto-inserted first or last child block, as the block might've been
+							// auto-inserted and then moved around a bit by the user.
+							autoInsertedBlock = innerBlocks.find(
+								( { name } ) => name === block.name
+							);
+						}
+
+						if ( autoInsertedBlock ) {
+							clientIds[ block.name ] =
+								autoInsertedBlock.clientId;
+						}
+
+						// TOOD: If no auto-inserted block was found in any of its designated locations,
```

Comment:
Are you planning on adding this one before merging?

Reply:
I was planning to do that in a follow-up. Would that be okay?

Reply:
Follow-up: #54024